src/asInput/index.jsx

```diff
@@ -11,6 +11,7 @@ export const getDisplayName = WrappedComponent =>
 export const inputProps = {
   label: PropTypes.string.isRequired,
   name: PropTypes.string.isRequired,
+  id: PropTypes.string,
```

Comment:
Does the `id` attribute have to be a `string`?  Should this also support `number`?

Reply:
`newId` returns a string and if were to come from the input prop I would expect something along the lines of `<InputText id="theID" ... />` and even if the ID there is `id={someID}` I think it would be cleaner to enforce that value to a `string`.